src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -3479,6 +3485,8 @@ HRESULT CorProfiler::CallTarget_RewriterCallback(RejitHandlerModule* moduleHandl
 
     hr = rewriter.Export();
 
+    delete[] newEHClauses;
```

Comment:
That could be risky no? you pass its pointer to the rewriter and ends it lifecyle before the rewriter lifecyle.
Couldn't you use `std::unique_ptr` instead ? (and passing the pointer to the rewriter using the `.get` method )?